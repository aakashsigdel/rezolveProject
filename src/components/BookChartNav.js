import React from 'react'
import { Link } from 'react-router'

import './BookChartNav.scss'

const BookChartNav = props =>
  <div className="nav-container">
    <Link to="/" activeClassName="nav-link-active" className="nav-link" onlyActiveOnIndex={true}>
      MY BOOKS
    </Link>
    <div className="nav-element-separator">&nbsp; | &nbsp;</div>
    <Link to="/charts" activeClassName="nav-link-active" className="nav-link">
      CHARTS
    </Link>
  </div>

export default BookChartNav
