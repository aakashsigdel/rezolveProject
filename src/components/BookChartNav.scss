@import '../variables';

.nav-container {
  display: flex;
  margin-top: 20px;
}

.nav-link {
  font-size: 18px;
  color: $nav_font;
}

.nav-link-active {
  font-size: 18px;
  font-weight: bold;
  color: $active;
}

.nav-element-separator {
  font-size: 18px;
  font-weight: bold;
  color: $nav_font;
}
