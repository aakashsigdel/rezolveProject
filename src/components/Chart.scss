@import '../variables';

.chart-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 30px;
}

.chart-label {
  margin-top: 10px;
  color: $active;
  font-size: 18px;
  font-weight: bold;
}
