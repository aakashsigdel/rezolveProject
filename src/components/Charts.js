import React from 'react'

const Charts = () => null

export default Charts
