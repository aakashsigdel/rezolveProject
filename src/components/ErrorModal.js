import React from 'react'
import { Link } from 'react-router'
import { connect } from 'react-redux'

import './ErrorModal.scss'

const ErrorModal = props =>
  <div className="error-container">
    <div className="error-heading">
      <div className="error"> Error </div>
      <div className="error-cross"> âœ• </div>
    </div>
    <div className="error-message">
      {props.message}
    </div>
    <div className="error-buttons">
      <Link to="/">
        <button className="nope" onClick={props.onCancel}>Cancel</button>
    </Link>
      <button className="ok" onClick={props.onReload}>Reload</button>
    </div>
  </div>

export default ErrorModal
