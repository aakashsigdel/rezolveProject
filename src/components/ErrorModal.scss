.error-container {
  position: absolute;
  height: 200px;
  color: #FFF;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background: #D45659;
  width: 500px;
  box-shadow: 0px 0px 10px 0px #352F3C;
  left: 60px;
  top: 60px;
}
.error-heading {
  font-size: 18px;
  padding: 0px 10px 0px 10px;
  align-items: center;
  flex: 1;
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid #C04C4F; 
}
.error-message {
  flex: 1.5;
  padding: 0px 10px 0px 10px;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 12px;
}
.error-buttons {
  padding-right: 10px;
  flex: 1.3;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  background: white;
}
.nope {
  width: 100px;
  height: 30px;
  border-style: none;
  border-radius: 5px;
  background: #BFBFBF;
  color: white;
  font-size: 18px;
}
.ok {
  font-size: 18px;
  margin: 0px 0px 0px 20px;
  color: white;
  width: 100px;
  height: 30px;
  border-style: none;
  border-radius: 5px;
  background: #DB585C;
}

