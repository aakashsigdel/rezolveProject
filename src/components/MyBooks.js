import React from 'react'
import { connect } from 'react-redux'
import { addBook } from '../actions/books'

import './MyBooks.scss'

export const titleFormat = title => title.length > 22 ? title.slice(0, 19) + '...' : title

export const Loader = () =>
  <div className="loader">
    <img src={'../../images/loading.gif'} className="loader-image" />
  </div>

export const EmptyResult = () =>
  <div className="books-container">
    <div className="book-not-found">
      0 books here! Search for a book from the search bar
    </div>
  </div>

const handleBookClick = props => {
  return props.searchText ? props.dispatch(addBook(props.book)) : null
}

export const Book = props => {
  const bookContainerClass = props.searchText
    ? "book-container book-container-overlay"
    : "book-container"
  return (
    <div className={bookContainerClass} onClick={() => handleBookClick(props)}>
      <img src={props.book.cover_image} className="book-thumbnail book-thumbnail-loading" />
      <div className="book-detail-container">
        <div className="book-title-container">
          <div className="book-title">{titleFormat(props.book.title)}</div>
          <img src="../../images/dots.png" className="book-title-dots" />
        </div>
        <div className="book-author">{props.book.author_name ? props.book.author_name : 'Not Found'}</div>
      </div>
    </div>
  )
}

export const MyBooks = props => {
  const books = props.params.searchText ? props.books : props.mybooks
  if (books.isLoading) {
    return <Loader />
  }

  if (books.books.length === 0) {
    return <EmptyResult />
  } else {
    return(
      <div className="books-container">
        {
          books.books.map(book =>
            <Book key={book.key} book={book}
              searchText={props.params.searchText}
              dispatch={props.dispatch}
            />)
        }
      </div>
    )
  }
}

const mapStateToProps = state => state

export default connect(mapStateToProps)(MyBooks)
