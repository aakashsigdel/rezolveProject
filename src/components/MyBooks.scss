@import '../variables';

.books-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  margin-top: 20px;
}

.loader {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
}

.loader-image {
  height: 80px;
  width: 80px;
}

.book-container {
  width: 180px;
  height: 350px;
}

.book-thumbnail {
  width: 180px;
  height: 298px;
}

.book-thumbnail-loading {
  background: transparent url('../../images/loading.gif') center no-repeat;
  background-size: 40px 40px;
}

.book-title-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-top: 10px;
}

.book-title {
  color: $primary_font;
  font-weight: bold;
}

.book-title-dots {
  width: 28px;
  height: 10px;
  align-self: center;
}

.book-author {
  color: $nav_font;
}
