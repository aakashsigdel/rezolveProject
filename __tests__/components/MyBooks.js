import React from 'react'
import { Book, EmptyResult, Loader, MyBooks, titleFormat } from '../../src/components/MyBooks'

import renderer from 'react-test-renderer'

export const books = {books: [
  {
    key: 111,
    author: 'John Doe',
    title: 'I am here',
    thumbnail_url: 'https://covers.openlibrary.org/b/id/295577-M.jpg'
  },
  {
    key: 112,
    author: 'Klio Author Sir',
    title: 'Phones will be always be gone',
    thumbnail_url: 'https://covers.openlibrary.org/b/id/295577-M.jpg'
  },
  {
    key: 113,
    author: 'Olivia Dolo',
    title: 'There will be water in the water',
    thumbnail_url: 'https://covers.openlibrary.org/b/id/295577-M.jpg'
  }
]}

describe('<MyBooks /> component', () => {
  it('should render <MyBooks /> correctly', () => {
    const tree = renderer.create(
      <MyBooks books={books} params={{searchText: "hello"}} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render <Book /> correctly', () => {
    const tree = renderer.create(
      <Book book={books.books[0]} searchText={'hello'} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render <Loader /> correctly', () => {
    const tree = renderer.create(<Loader />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render <EmptyResult /> correctly', () => {
    const tree = renderer.create(<EmptyResult />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should format title correctly', () => {
    const title = 'this is a long string with more than 22 chars'
    const expectedTitle = title.slice(0, 19) + '...'

    expect(titleFormat(title)).toEqual(expectedTitle)
  })
})
