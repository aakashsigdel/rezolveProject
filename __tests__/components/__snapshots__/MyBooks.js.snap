exports[`<MyBooks /> component should render <Book /> correctly 1`] = `
<div
  className="book-container book-container-overlay"
  onClick={[Function]}>
  <img
    className="book-thumbnail book-thumbnail-loading"
    src={undefined} />
  <div
    className="book-detail-container">
    <div
      className="book-title-container">
      <div
        className="book-title">
        I am here
      </div>
      <img
        className="book-title-dots"
        src="../../images/dots.png" />
    </div>
    <div
      className="book-author">
      Not Found
    </div>
  </div>
</div>
`;

exports[`<MyBooks /> component should render <EmptyResult /> correctly 1`] = `
<div
  className="books-container">
  <div
    className="book-not-found">
    0 books here! Search for a book from the search bar
  </div>
</div>
`;

exports[`<MyBooks /> component should render <Loader /> correctly 1`] = `
<div
  className="loader">
  <img
    className="loader-image"
    src="../../images/loading.gif" />
</div>
`;

exports[`<MyBooks /> component should render <MyBooks /> correctly 1`] = `
<div
  className="books-container">
  <div
    className="book-container book-container-overlay"
    onClick={[Function]}>
    <img
      className="book-thumbnail book-thumbnail-loading"
      src={undefined} />
    <div
      className="book-detail-container">
      <div
        className="book-title-container">
        <div
          className="book-title">
          I am here
        </div>
        <img
          className="book-title-dots"
          src="../../images/dots.png" />
      </div>
      <div
        className="book-author">
        Not Found
      </div>
    </div>
  </div>
  <div
    className="book-container book-container-overlay"
    onClick={[Function]}>
    <img
      className="book-thumbnail book-thumbnail-loading"
      src={undefined} />
    <div
      className="book-detail-container">
      <div
        className="book-title-container">
        <div
          className="book-title">
          Phones will be alwa...
        </div>
        <img
          className="book-title-dots"
          src="../../images/dots.png" />
      </div>
      <div
        className="book-author">
        Not Found
      </div>
    </div>
  </div>
  <div
    className="book-container book-container-overlay"
    onClick={[Function]}>
    <img
      className="book-thumbnail book-thumbnail-loading"
      src={undefined} />
    <div
      className="book-detail-container">
      <div
        className="book-title-container">
        <div
          className="book-title">
          There will be water...
        </div>
        <img
          className="book-title-dots"
          src="../../images/dots.png" />
      </div>
      <div
        className="book-author">
        Not Found
      </div>
    </div>
  </div>
</div>
`;
